test/common.py

```diff
@@ -623,6 +623,9 @@ def metafunc(self, standalone):
       if not standalone:
         func(self)
       else:
+        nonlocal exclude_engines
+        if exclude_engines is None:
+          exclude_engines = []
```

Comment:
I think you just put these two lines at the top of the function (before the `def decorated`) and avoid the nonlocal thing.

You could also just write `exclude_engines=[]` in the signature and add `# noqa` to silence the warning.

Reply:
done! `exclude_engines` isn't modified so doing it in the signature sounds good.